satpy/readers/seviri_l2_bufr.py

```diff
@@ -53,9 +55,22 @@
 
 
 class SeviriL2BufrFileHandler(BaseFileHandler):
-    """File handler for SEVIRI L2 BUFR products."""
+    """File handler for SEVIRI L2 BUFR products.
+
+    **Loading data as AreaDefinition**
+
+    By providing the `as_area_def` as True in the `reader_kwargs`, the dataset is loaded as
+    an AreaDefinition using a standardized AreaDefinition in areas.yaml. By default, the
+    dataset will be loaded as a SwathDefinition, i.e. similar to how the data are stored in
+    in the BUFR file:
+
+        scene = satpy.Scene(filenames,
+                            reader='seviri_l2_bufr',
+                            reader_kwargs={'as_area_def': False})
+
+    """
 
-    def __init__(self, filename, filename_info, filetype_info, **kwargs):
+    def __init__(self, filename, filename_info, filetype_info, as_area_def=False, **kwargs):
```

Comment:
Are there any times when trying to produce the AreaDefinition will fail? Or produce wrong output? What if you didn't have a kwarg and always tried to produce an AreaDefinition?

Reply:
The thing is that the data are stored as lat/lon pairs in the file without any projection information, so I want the default behavior to be to return the data as it's stored in the file, and hence as a `SwathDefinition` (this is also how the reader worked before adding this feature). Therefore I think it's better to stick to the kwarg and only produce an `AreaDefinition` if the user asks for it.

In principle the code should not fail when producing the `AreaDefinition` as it just puts the data points on a predefined grid using the corresponding lat/lon pairs. I did, however, just implement a fix that allows the code to handle situations where the dataset contains data points outside the predefined `AreaDefinition` (before the code would have failed in such situations).
, And then add the possibility to put the lat/lon data on a pre-defined grid using the kwarg